zk_toolbox/crates/zk_inception/src/commands/portal.rs

```diff
@@ -0,0 +1,178 @@
+            },
+        })
+    } else {
+        None
+    };
+    // Base token:
+    let (base_token_addr, base_token_info) = if chain_config.base_token == BaseToken::eth() {
+        (
+            ethers::utils::hex::encode_prefixed(Address::zero()),
+            TokenInfo {
+                symbol: "ETH".to_string(),
+                name: "Ether".to_string(),
+                decimals: 18,
+            },
+        )
+    } else {
+        (
+            ethers::utils::hex::encode_prefixed(chain_config.base_token.address),
+            ethereum::get_token_info(chain_config.base_token.address, l1_rpc_url.to_string())
+                .await?,
+        )
+    };
+    // Tokens: add base token
+    let tokens = vec![TokenConfig {
+        address: L2_BASE_TOKEN_ADDRESS.to_string(),
+        l1_address: Some(base_token_addr.to_string()),
+        symbol: base_token_info.symbol,
+        decimals: base_token_info.decimals,
+        name: Some(base_token_info.name.to_string()),
+    }];
+    // Build hyperchain config
+    Ok(HyperchainConfig {
+        network: NetworkConfig {
+            id: chain_config.chain_id.as_u64(),
+            key: chain_config.name.clone(),
+            name: chain_config.name.clone(),
+            rpc_url: rpc_url.to_string(),
+            public_l1_network_id,
+            l1_network,
+            block_explorer_url: None,
+            block_explorer_api: None,
+        },
+        tokens,
+    })
+}
+
+pub async fn create_hyperchains_config(
+    chain_configs: &[ChainConfig],
+) -> anyhow::Result<HyperchainsConfigs> {
+    let mut hyperchain_configs = Vec::new();
+    for chain_config in chain_configs {
+        match create_hyperchain_config(chain_config).await {
+            Ok(config) => hyperchain_configs.push(config),
+            Err(_) => logger::warn(msg_portal_failed_to_configure_chain(&chain_config.name)),
+        }
+    }
+    // Ensure at least one HyperchainConfig is created
+    if hyperchain_configs.is_empty() {
+        Err(anyhow!(MSG_FAILED_TO_START_PORTAL_ERR))
+    } else {
+        Ok(HyperchainsConfigs(hyperchain_configs))
+    }
+}
+
+pub async fn run(shell: &Shell, args: PortalArgs) -> anyhow::Result<()> {
+    let ecosystem_config = EcosystemConfig::from_file(shell)?;
+    let chains = ecosystem_config.list_of_chains();
+
+    let mut chain_configs = Vec::new();
+    for chain in chains {
+        match ecosystem_config.load_chain(Some(chain.clone())) {
+            Some(chain_config) => chain_configs.push(chain_config),
+            None => {
+                logger::warn(msg_portal_chain_not_initialized(&chain));
+                continue;
+            }
+        }
+    }
+    if chain_configs.is_empty() {
+        return Err(anyhow!(MSG_FAILED_TO_START_PORTAL_ERR));
+    }
+
+    let hyperchain_configs = create_hyperchains_config(&chain_configs).await?;
+    let runtime_config = PortalRuntimeConfig {
+        node_type: "hyperchain".to_string(),
+        hyperchain_configs,
+    };
```

Comment:
should we move it to the `init` process and regenerate it only when we add a new chain?

Reply:
moved